core/dbt/events/types.py

```diff
@@ -416,6 +416,21 @@ def message(self):
         return line_wrap_message(warning_tag(f"Deprecated functionality\n\n{description}"))
 
 
+@dataclass
+class FunctionDeprecated(WarnLevel, pt.FunctionDeprecated):
+    def code(self):
+        return "D008"
+
+    def message(self):
+        description = (
+            f"Call to deprecated function {self.function_name}. \n"
+            f"Details of deprecation: {self.reason} \n"
+            f"To resolve this deprecation: {self.suggested_action} \n"
+            f"This will be removed in version: {self.version}."
+        )
+        return line_wrap_message(warning_tag(f"Deprecated functionality\n\n{description}"))
+
```

Comment:
This is what the text log line looks like:
```
[0m22:33:56.723169 [warn ] [MainThread]: [[33mWARNING[0m]: Deprecated functionality
Call to deprecated function raise_compiler_error.  Details of deprecation: Not
doing it this way  To resolve this deprecation: raise CompilationException
directly  This will be removed in version: 1.5.0.
```

Reply:
@dbeatty10 can I get your feedback on what information would be useful to get passed as an adapter maintainer?  The _user_ won't be able to do anything here but ideally the maintainer can easily fix the deprecated function.

Reply:
Sorry for my late reply here @emmyoop !

Here's a variant aiming to be succint and readable:
```
17:42:33  [WARNING]: `raise_compiler_error` is deprecated and will be removed in dbt-core version 1.5.0

Adapter maintainers can resolve this deprecation by using `raise CompilationException` directly instead.
Lorem ipsum dolor sit amet
```

Here's where each of the relevant variables would fit in:
```
17:42:33  [WARNING]: `{self.function_name}` is deprecated and will be removed in dbt-core version {self.version}

Adapter maintainers can resolve this deprecation by {self.suggested_action}.
{self.reason}
```